psyneulink/core/components/functions/nonstateful/combinationfunctions.py

```diff
@@ -723,19 +725,25 @@ class Parameters(CombinationFunction.Parameters):
         offset = Parameter(0.0, modulable=True, aliases=[ADDITIVE_PARAM])
         changes_shape = Parameter(True, stateful=False, loggable=False, pnl_internal=True)
 
-    @tc.typecheck
+    @beartype
     def __init__(self,
-                 # weights: tc.optional(parameter_spec)=None,
-                 # exponents: tc.optional(parameter_spec)=None,
+                 # weights:  Optional[ValidParamSpecType] = None,
+                 # exponents:  Optional[ValidParamSpecType] = None,
                  weights=None,
                  exponents=None,
                  default_variable=None,
-                 operation: tc.optional(tc.enum(SUM, PRODUCT)) = None,
-                 scale: tc.optional(parameter_spec) = None,
-                 offset: tc.optional(parameter_spec) = None,
+                 operation: Optional[Literal['sum', 'product']] = None,
```

Comment:
Does this work with `kw.SUM`/`kw.PRODUCT` ?

Reply:
Do you mean indexing typing.Literal with kw.SUM? Unfortunately no, since they are not defined statically. If we made all keywords enums then you CAN index Literal with them it seems. I was thinking of doing this but I couldn't come up with something that didn't introduce a ton of code changes. They probably should be enums though. One way that might work would be to move them all to enums and then programmatically instantiate the keyword globals from the enums.

Reply:
yeah, having keywords as enums would be nice, but that's not a good fit for this change.
would it work if keywords were typed hinted at definition, something like `SUM:Literal['sum'] = 'sum'`?
are there many keywords that would need such treatment?

Reply:
ok @jvesely and @jdcpni,

something like this seems to work:

```
from beartype import beartype
from typing_extensions import Literal, Final
from enum import Enum


class Keyword(Enum):
    SUM = "sum"
    PRODUCT = "product"


SUM: Final = Keyword.SUM
PRODUCT: Final = Keyword.PRODUCT


@beartype
def test_literal(b: Literal[Keyword.SUM, Keyword.PRODUCT]) -> None:
    print(b)


# Passes runtime and static type checks.
test_literal(SUM)
test_literal(PRODUCT)
test_literal(Keyword.SUM)
test_literal(Keyword.PRODUCT)

# Fails runtime type check
test_literal("sum")
```

This would require having two lines in keywords.py for each keyword. One for the enum, and one for the module constant variable for that keyword. So if someone wants to add a keyword, they would need to know about this and put it in both places. If this looks good, I can try to implement it like this. I anticipate the there are some places in the code that expect these keywords to actually be strings which is no longer the case using this approach. This can probably be fixed by simply using `SUM.value` in those places though.

What do you both think?